src/Microsoft.IdentityModel.Xml/XmlUtil.cs

```diff
@@ -93,6 +94,71 @@ public static void CheckReaderOnEntry(XmlReader reader, string element, string @
         }
 
         /// <summary>
+        /// Checks if the <see cref="XmlReader"/> is pointing to an expected element.
+        /// </summary>
+        /// <param name="reader">the <see cref="XmlReader"/>to check.</param>
+        /// <param name="element">the expected element.</param>
+        /// <param name="namespaceList">the expected namespace list.</param>
+        /// <exception cref="ArgumentNullException">if <paramref name="reader"/> is null.</exception>
+        /// <exception cref="ArgumentNullException">if <paramref name="element"/> is null or empty.</exception>
+        /// <exception cref="XmlReadException">if <paramref name="reader"/> if not at a StartElement.</exception>
+        /// <exception cref="XmlReadException">if <paramref name="reader"/> if not at expected element.</exception>
+        public static void CheckReaderOnEntry(XmlReader reader, string element, ICollection<string> namespaceList)
```

Comment:
Can we make this internal? If we ever support WsTrust we would handle multiple versions in a different way. For plucking a token out of a WsFed message, it seems fine.